x-pack/plugins/actions/server/application/connector/schemas/connector_schema.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const connectorSchema = schema.object({
+  id: schema.string(),
+  name: schema.string(),
+  actionTypeId: schema.string(),
+  config: schema.maybe(schema.recordOf(schema.string(), schema.any())),
```

Comment:
Should we get the config schema from the connectorType (validate.config.schema) and use here?

Reply:
It would be worth exploring as a follow up. We may need a fn like `getConnectorSchema` where it returns this schema with the config and secrets having proper validation.